resources/todomvc/architecture-examples/react/package.json

```diff
@@ -1,20 +1,40 @@
 {
-  "private": true,
-  "scripts": {
-    "build": "babel --plugins transform-react-jsx js/app.jsx | uglifyjs -o build.min.js"
-  },
-  "dependencies": {
-    "classnames": "^2.2.5",
-    "director": "^1.2.8",
-    "react": "15.5.4",
-    "react-dom": "15.5.4",
-    "todomvc-app-css": "^2.0.0",
-    "todomvc-common": "^1.0.2"
-  },
-  "devDependencies": {
-    "babel-cli": "^6.10.1",
-    "babel-plugin-transform-react-jsx": "^6.8.0",
-    "babel-plugin-uglify": "^1.0.2",
-    "uglify-js": "^2.6.4"
-  }
+    "name": "library-react",
+    "version": "1.0.0",
+    "description": "TodoMVC app written with React.",
+    "private": true,
+    "engines": {
+        "node": ">=18.13.0",
+        "npm": ">=8.19.3"
+    },
+    "scripts": {
+        "start": "webpack serve --open --config webpack.dev.js",
+        "build": "webpack --config webpack.prod.js",
+        "serve": "http-server ./dist -p 7002 -c-1 --cors",
+        "test": "jest"
+    },
+    "devDependencies": {
+        "@babel/core": "^7.21.0",
+        "@babel/preset-env": "^7.20.2",
+        "@babel/preset-react": "^7.18.6",
+        "babel-loader": "^9.1.2",
+        "css-loader": "^6.7.3",
+        "css-minimizer-webpack-plugin": "^4.2.2",
+        "eslint-plugin-react": "^7.32.2",
+        "html-webpack-plugin": "^5.5.0",
+        "http-server": "^14.1.1",
+        "mini-css-extract-plugin": "^2.7.2",
+        "style-loader": "^3.3.1",
+        "webpack": "^5.75.0",
+        "webpack-cli": "^5.0.1",
+        "webpack-dev-server": "^4.11.1",
+        "webpack-merge": "^5.8.0"
+    },
+    "dependencies": {
+        "classnames": "^2.2.5",
+        "react": "^17.0.2",
```

Comment:
Apologies for the drive-by question: but why pick v17 and not v18 (current stable)?

Reply:
I believe there are 2 reasons:
* v17 is still the most used version in the web currently.
* v18 is much more asynchronous, so it's difficult (if possible) to determine when it finished rendering. We'll have to eventually solve this problem though...

Reply:
> * so it's difficult (if possible) to determine when it finished rendering. We'll have to eventually solve this problem though...

At API Level, React ~v17 also has similar problem that is hard to know when the "rendering" finished and the 3rd argument of [`render()`](https://react.dev/reference/react-dom/render) has been used for that. 

After v18, these guide might be helpful

- https://github.com/reactwg/react-18/discussions/5
- https://react.dev/blog/2022/03/08/react-18-upgrade-guide

Reply:
FYI: Some more context on why v17 was chosen is in this [document](https://docs.google.com/document/d/1MXxk9DryJmvAmPMt2uMneaS3J5Y-dMFGt4EGvMzhhJE/edit?disco=AAAAsVSN7aM) that Thorsten put together.

Reply:
+1 to what @gsathya mentioned. NPM data suggests that 18 is already the most popular version, and in-the-wild numbers will naturally trail installs. It's really important that we're all optimizing for the right version and I'm very concerned about the choice of 17 here. There is a very big difference between 17 and 18, and optimizations that benefit one may not help the other.

If the question is about how to measure accurately in 18, we can help with that :-)

Reply:
Thanks for the useful feedback, folks. I agree that v18 will likely have different performance characterstics compared to v17. It would make sense to include a version of the benchmark with v18.
But also please note that we also intend to update the benchmark more often than what we did so far.